Content.Server/Abilities/Mime/MimePowersSystem.cs

```diff
@@ -0,0 +1,144 @@
+        {
+            base.Update(frameTime);
+            foreach (var entity in RemQueue)
+            {
+                RemComp<VowbreakerComponent>(entity);
+            }
+            RemQueue.Clear();
+            /// Queue to despawn invis walls
+            foreach (var invisWall in EntityQuery<InvisibleWallComponent>())
+            {
+                invisWall.Accumulator += frameTime;
+                if (invisWall.Accumulator < invisWall.DespawnTime)
+                {
+                    continue;
+                }
+                EntityManager.QueueDeleteEntity(invisWall.Owner);
+            }
+            /// Queue to track whether mimes can retake vows yet
+            foreach (var (vowbreaker, mime) in EntityQuery<VowbreakerComponent, MimePowersComponent>())
+            {
+                vowbreaker.Accumulator += frameTime;
+                if (vowbreaker.Accumulator < vowbreaker.VowCooldown)
+                {
+                    continue;
+                }
+                mime.ReadyToRepent = true;
+                _popupSystem.PopupEntity(Loc.GetString("mime-ready-to-repent"), mime.Owner, Filter.Entities(mime.Owner));
+                RemQueue.Enqueue(vowbreaker.Owner);
+            }
+        }
+
+        private void OnComponentInit(EntityUid uid, MimePowersComponent component, ComponentInit args)
+        {
+            _actionsSystem.AddAction(uid, component.InvisibleWallAction, uid);
+            _alertsSystem.ShowAlert(uid, AlertType.VowOfSilence);
+        }
+        private void OnSpeakAttempt(EntityUid uid, MimePowersComponent component, SpeakAttemptEvent args)
+        {
+            if (!component.Enabled)
+                return;
+
+            _popupSystem.PopupEntity(Loc.GetString("mime-cant-speak"), uid, Filter.Entities(uid));
+            args.Cancel();
+        }
+
+        /// <summary>
+        /// Creates an invisible wall in a free space after some checks.
+        /// </summary>
+        private void OnInvisibleWall(EntityUid uid, MimePowersComponent component, InvisibleWallActionEvent args)
+        {
+            if (!component.Enabled)
+                return;
+
+            var xform = Transform(uid);
+            /// Get the tile in front of the mime
+            var offsetValue = xform.LocalRotation.ToWorldVec().Normalized;
+            var coords = xform.Coordinates.Offset(offsetValue).SnapToGrid();
+            /// Check there are no walls or mobs there
+            foreach (var entity in coords.GetEntitiesInTile())
+            {
+                if (_tagSystem.HasTag(entity, "Wall") || (HasComp<MobStateComponent>(entity) && entity != uid) ||
+                    (TryComp<DoorComponent>(entity, out var door) && door.State == DoorState.Closed))
+                {
+                    _popupSystem.PopupEntity(Loc.GetString("mime-invisible-wall-failed"), uid, Filter.Entities(uid));
+                    return;
+                }
+            }
```

Comment:
feels like it should just check
1) that there are no mobs
2) that there are no entities with an impassible fixture